ores/lock_manager/poolcounter.py

```diff
@@ -4,31 +4,54 @@
 This is useful to make sure too many connections are not coming from one IP.
 """
 
+import logging
 import socket
+import hashlib
 
 from .lock_manager import LockManager
+from ..errors import TimeoutError
+
+logger = logging.getLogger(__name__)
 
 
 class PoolCounter(LockManager):
-    def __init__(self, config):
-        self.server = config['server']
-        self.port = config['port']
+    def __init__(self, nodes):
+        self.nodes = nodes
         self.stream = None
 
-    def connect(self):
-        self.stream = socket.socket(socket.AF_INET, socket.SOCK_STREAM, 0)
-        return self.stream.connect((self.server, self.port))
+    def connect(self, key):
+        hashes = []
+        for node in self.nodes:
+            hashes.append(
+                (node, hashlib.sha256(bytes(node[0] + key, 'utf-8')).hexdigest())
+            )
+        node = sorted(hashes, key=lambda i: i[1])[0][0]
+        try:
+            self.stream = socket.socket(socket.AF_INET, socket.SOCK_STREAM, 0)
+            self.stream.connect(node)
+        except ConnectionRefusedError:
+            logger.warning(
+                'Can not connect to the PoolCounter node %s' % node[0])
+            return False
```

Comment:
We could also fall back to the next server, but I'm sure this is fine for now.

Reply:
we return false and upstream service just ignores that.

Reply:
Alright!